generator/src/lib.rs

```diff
@@ -85,6 +85,49 @@ named!(cfloat<&str, f32>,
     terminated!(nom::float_s, char!('f'))
 );
 
+pub fn define_test() -> Tokens {
+    quote! {
+        #[cfg(test)]
+        mod tests {
+            use vk;
+            #[test]
+            fn test_ptr_chains() {
+                unsafe fn ptr_chain_iter<T>(
+                    ptr: &mut T,
+                ) -> impl Iterator<Item = *mut vk::BaseOutStructure> {
+                    use std::ptr::null_mut;
+                    let ptr: *mut vk::BaseOutStructure = ptr as *mut T as _;
+                    let ptr: *mut vk::BaseOutStructure = (*ptr).p_next;
+                    (0..).scan(ptr, |p_ptr, _| {
+                        if *p_ptr == null_mut() {
+                            return None;
+                        }
+                        let n_ptr = (**p_ptr).p_next as *mut vk::BaseOutStructure;
+                        let old = *p_ptr;
+                        *p_ptr = n_ptr;
+                        Some(old)
+                    })
+                }
+                let mut variable_pointers = vk::PhysicalDeviceVariablePointerFeatures::builder();
+                let mut corner =
+                    vk::PhysicalDeviceCornerSampledImageFeaturesNV::builder();
+                let chain = vec![
+                    &variable_pointers as *const _ as usize,
+                    &corner as *const _ as usize,
+                ];
+                let mut device_create_info = vk::DeviceCreateInfo::builder()
+                    .push_next(&mut corner)
+                    .push_next(&mut variable_pointers);
+                let chain2: Vec<usize> = unsafe {
+                    ptr_chain_iter(&mut device_create_info)
+                        .map(|ptr| ptr as usize)
+                        .collect()
+                };
+                assert_eq!(chain, chain2);
+            }
+        }
+    }
+}
```

Comment:
Does this need to be in the generated code, rather than a regular module of the `ash` crate?

Reply:
We could have this code outside of the generator, and if we have `ash-sys` it will then be part of that crate. I'll change it.